flink-connector-opensearch/src/test/java/org/apache/flink/streaming/connectors/opensearch/OpensearchSinkTest.java

```diff
@@ -79,7 +82,15 @@ public void setUp() throws IOException {
                         return (req, resp, context) -> resp.setStatusCode(200);
                     } else if (method.equalsIgnoreCase("POST")) {
                         // Bulk responses are configured per test case
-                        return (req, resp, context) -> responses.poll().accept(resp);
+                        return (req, resp, context) -> {
+                            lock.lock();
+                            try {
+                                responses.poll().accept(resp);
+                                flushed.signalAll();
```

Comment:
Or you could just use a OneShotLatch which makes all of this logic here a one-liner _and_ gives you more control to wait for a specific message to be consumed :shrug: 
(and you wouldn't need `awaitForCondition`)

Reply:
`awaitForCondition` is not checking the flushing part, it is checking the `numPendingRequest`, which will be eventually updated